src/components/Settings/Speech/Speech.container.js

```diff
@@ -1,169 +1,125 @@
-
-import FullScreenDialog from '../../FullScreenDialog';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import debounce from 'lodash.debounce';
+import { injectIntl } from 'react-intl';
+
 import {
-  MIN_PITCH,
-  MAX_PITCH,
-  INCREMENT_PITCH,
-  MIN_RATE,
-  MAX_RATE,
-  INCREMENT_RATE
-} from './Speech.constants';
+  speak,
+  changeVoice,
+  changePitch,
+  changeRate
+} from '../../SpeechProvider/SpeechProvider.actions';
+import SpeechComponent from './Speech.component';
 import messages from './Speech.messages';
 
-const styles = theme => ({
-  container: {
-    display: 'flex',
-    position: 'relative',
-    justifyContent: 'center'
-  },
-  icon: {
-    marginLeft: 3,
-    marginRight: 3,
-    width: 20
-  },
-  progress: {
-    paddingLeft: 2,
-    paddingRight: 2,
-    maxWidth: '50%',
-    minWidth: '20%',
-    marginTop: 20
+export class Speech extends Component {
+  static propTypes = {
+    /**
+     * If true, Speech will be visible
+     */
+    open: PropTypes.bool,
+    /**
+     * Active language
+     */
+    lang: PropTypes.string,
+    speech: PropTypes.object,
+    voices: PropTypes.array,
+    onRequestClose: PropTypes.func,
+    changeVoice: PropTypes.func,
+    changePitch: PropTypes.func,
+    changeRate: PropTypes.func
+  };
+
+  state = {
+    selectedVoiceIndex: 0,
+    voiceOpen: false,
+    anchorEl: null
+  };
+
+  speakSample = debounce(() => {
+    const { intl, speak } = this.props;
+
+    const text = intl.formatMessage(messages.sampleSentence);
+    speak(text);
+  }, 500);
+
+  handleClickListItem = event => {
+    this.setState({ voiceOpen: true, anchorEl: event.currentTarget });
+  };
+
+  handleMenuItemClick = ({ voiceURI, lang }, index) => {
+    const { changeVoice } = this.props;
+
+    changeVoice(voiceURI, lang);
+    this.speakSample();
+    this.setState({ voiceOpen: false, selectedVoiceIndex: index });
+  };
+
+  handleChangePitch = value => {
+    const { changePitch } = this.props;
+
+    changePitch(value);
+    this.speakSample();
+  };
+
+  handleChangeRate = value => {
+    const { changeRate } = this.props;
+
+    changeRate(value);
+    this.speakSample();
+  };
+
+  handleVoiceRequestClose = () => {
+    this.setState({ voiceOpen: false });
+  };
+
+  render() {
+    const {
+      intl,
+      open,
+      lang,
+      onRequestClose,
+      speech: { voices, options: { voiceURI, pitch, rate } }
+    } = this.props;
+
+    const langVoices = voices.filter(
+      voice => voice.lang.slice(0, 2) === lang.slice(0, 2)
+    );
+
+    return (
+      <SpeechComponent
+        {...this.state}
+        handleChangePitch={this.handleChangePitch}
+        handleChangeRate={this.handleChangeRate}
+        handleClickListItem={this.handleClickListItem}
+        handleMenuItemClick={this.handleMenuItemClick}
+        handleVoiceRequestClose={this.handleVoiceRequestClose}
+        intl={intl}
```

Comment:
I wouldn't pass `intl` as prop, instead I would add the dependency within `SpeechComponent`:

```
import { injectIntl, intlShape } from 'react-intl';
```

Reply:
But wouldn't that be redundant? `SpeechContainer` uses `intl` in the `speakSample` function, so I think it makes more sense to just pass down the props instead of wrapping `SpeechComponent` with `injectIntl` again.

Reply:
You are right, I was going to defend my position but I see your point 👍

Reply:
Yes, I understand that. If you have any other approach in mind, let me know, please.